README.md

```diff
@@ -382,151 +424,454 @@ Jest uses Jasmine 2 by default. An introduction to Jasmine 2 can be found
+
+```js
+describe('ounces per can', () => {
+  it('is less than 20', () => {
+    expect(ouncesPerCan()).toBeLessThan(10);
+  });
+});
+```
+
+### `.toBeLessThanOrEqual(number)`
+
+To compare floating point numbers, you can use `toBeLessThanOrEqual`. For example, if you want to test that `ouncesPerCan()` returns a value of at most 12 ounces, write:
+
+```js
+describe('ounces per can', () => {
+  it('is at most 12', () => {
+    expect(ouncesPerCan()).toBeLessThanOrEqual(12);
+  });
+});
+```
+
+### `.toBeNull()`
+
+`.toBeNull()` is the same as `.toBe(null)` but the error messages are a bit nicer. So use `.toBeNull()` when you want to check that something is null.
+
+```js
+function bloop() {
+  return null;
+}
+
+describe('bloop', () => {
+  it('returns null', () => {
+    expect(bloop()).toBeNull();
+  });
+})
+```
+
+### `.toBeTruthy()`
+
+Use `.toBeTruthy` when you don't care what a value is, you just want to ensure a value is true in a boolean context. For example, let's say you have some application code that looks like:
+
+```js
+drinkSomeLaCroix();
+if (thirstInfo()) {
+  drinkMoreLaCroix();
+}
+```
+
+You may not care what `thirstInfo` returns, specifically - it might return `true` or a complex object, and your code would still work. So if you just want to test that `thirstInfo` will be truthy after drinking some La Croix, you could write:
+
+```js
+describe('drinking some La Croix', () => {
+  it('leads to having thirst info', () => {
+    drinkSomeLaCroix();
+    expect(thirstInfo()).toBeTruthy();
+  });
+});
+```
+
+In JavaScript, there are six falsy values: `false`, `0`, `''`, `null`, `undefined`, and `NaN`. Everything else is truthy.
+
+### `.toBeUndefined()`
+
+Use `.toBeUndefined` to check that a variable is undefined. For example, if you want to check that a function `bestDrinkForFlavor(flavor)` returns `undefined` for the `'octopus'` flavor, because there is no good octopus-flavored drink:
+
+```js
+describe('the best drink', () => {
+  it('for octopus flavor is undefined', () => {
+    expect(bestDrinkForFlavor('octopus')).toBeUndefined();
+  });
+});
+```
+
+You could write `expect(bestDrinkForFlavor('octopus')).toBe(undefined)`, but it's better practice to avoid referring to `undefined` directly in your code.
+
+### `.toContain(item)`
 
-## Mock API
+Use `.toContain` when you want to check that an item is in a list.
+
+For example, if `getAllFlavors()` returns a list of flavors and you want to be sure that `lime` is in there, you can write:
+
+```js
+describe('the list of all flavors', () => {
+  it('contains lime', () => {
+    expect(getAllFlavors()).toContain('lime');
+  });
+});
+```
+
+### `.toEqual(value)`
+
+Use `.toEqual` when you want to check that two objects are equal when you recursively check for the equality of all fields, rather than checking for object identity. For example, `toEqual` and `toBe` behave differently in this test suite, so all the tests pass:
```

Comment:
I'm not sure I understand this sentence. Maybe:

"Use `.toEqual` when you want to check that two objects are equal. This matcher recursively checks for the equality of all fields, rather than checking for object identity."


One difference of this matcher in Jest compared to Jasmine 2 is that it ignores `undefined` fields in either objects and doesn't compare the constructor names. At FB we have more than 1000 individual tests relying on this Jasmine 1 behavior so we patched our Jasmine fork for this.

We may in the future opt to add `toStrictEqual`.